core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts

```diff
@@ -151,16 +154,31 @@ export class CkEditorInitializerService {
                   },
                   function(widgetShouldBeRemoved) {
                     if (widgetShouldBeRemoved || that.data.isCopied) {
+                      const highlighted = customizationArgSpecs
+                        .some(function(spec) {
+                          return spec.default_value_obtainable_from_highlight;
+                        });
+
                       that.data.isCopied = false;
                       var newWidgetSelector = (
                         '[data-cke-widget-id="' + that.id + '"]');
-                      if (newWidgetSelector !== null) {
-                        var widgetElement = editor.editable().findOne(
-                          newWidgetSelector);
-                        if (widgetElement) {
-                          widgetElement.remove();
-                          editor.fire('change');
-                        }
+                      if (newWidgetSelector === null) {
+                        return;
+                      }
+                      var widgetElement = editor.editable().findOne(
+                        newWidgetSelector);
+
+                      if (!widgetElement) {
+                        return;
+                      }
+
+                      if ((!highlighted && componentIsNewlyCreated) ||
+                           !componentIsNewlyCreated) {
+                        widgetElement.remove();
+                        editor.fire('change');
+                      } else {
+                        editor.loadSnapshot(that.initialSnapshot);
```

Comment:
What if we load snapshot in all situations? I mean simply
```
if (widgetElement) {
  wdigetElement.remove();
  editor.loadSnapshot(that.initialSnapshot);
  editor.fire('change');
}
```

Will it work?

Reply:
Maybe, but I believe this calls for the delete functionality too so remove would come after loadSnapshot, but I separated it too make more sense.